pkg/tar/tar_test.go

```diff
@@ -167,119 +168,126 @@ func checkExpectedFiles(dir string, expectedFiles map[string]*fileInfo) error {
 }
 
 func TestExtractTarFolders(t *testing.T) {
-	if !sys.HasChrootCapability() {
-		t.Skipf("chroot capability not available. Disabling test.")
-	}
+	for _, insecure := range []bool{false, true} {
+		if !insecure && !sys.HasChrootCapability() {
+			t.Logf("chroot capability not available. Skipping part of test.")
```

Comment:
I think splitting these out is preferable, they're quite distinct paths conceptually:
```
func TestExtractTarFoldersChroot(t *testing.T) {
  if !sys.HasChrootCapability() {
    t.Skipf(...)
  }
  testExtractTarFolders(t, true) 
}

func TestExtractTarFoldersInsecure(t *testing.T) {
  testExtractTarFolders(t, false)
}

func testExtractTarFolders(t *testing.T, chroot bool) 
```

(The bool is kinda icky, ideally testExtractTarFolders could have a nicer signature with some kind of func() but I'm not sure what it looks like off the top of my head)